PWGHF/D2H/Tasks/taskDplus.cxx

```diff
@@ -258,11 +393,52 @@ struct HfTaskDplus {
     registry.fill(HIST("hEtaGen"), particle.eta());
   }
 
+  // Fill THnSparse of quantities for generated Dplus particles
+  /// \param particle is a particle with MC information
+  /// \param ptGenB transverse momentum of beauty mother for nonprompt candidates
+  /// \param flagGenB transverse momentum of beauty mother for nonprompt candidates
+  /// \param centrality collision centrality
+  /// \param occupancy collision occupancy
+  template <typename T1>
+  void fillSparseMCGen(const T1& particle,
+                       float& ptGenB,
+                       int& flagGenB,
+                       float& centrality,
+                       float& occupancy)
```

Comment:
Passing variables by reference allows the function to change their values which is probably not what you want.